scss/forms/_star-rating.scss

```diff
@@ -37,44 +36,85 @@
   }
 
   &:hover input + label {
-    @include form-star-rating($black);
+    @include form-star-rating($black, $black);
   }
 
   > input:hover ~ label {
-    @include form-star-rating($gray-500);
+    @include form-star-rating($gray-700, $white);
   }
 
   > input:hover + label {
-    @include form-star-rating($black);
+    @include form-star-rating($black, $black);
+  }
+
+  &.disabled,
+  &:disabled {
+    pointer-events: none;
+
+    > label {
+      @include form-star-rating($gray-700, $gray-700);
+    }
+
+    > input:checked ~ label {
+      @include form-star-rating($gray-500, $gray-500);
+    }
+
+    > input:checked + label {
+      @include form-star-rating($gray-700, $gray-700);
+    }
   }
 }
 
 .star-rating-dark {
   > label {
-    @include form-star-rating($brand-orange);
+    @include form-star-rating($brand-orange, $brand-orange);
   }
 
   > input:checked ~ label {
-    @include form-star-rating($gray-700);
+    @include form-star-rating($gray-500, $black);
   }
 
   > input:checked + label {
-    @include form-star-rating($brand-orange);
+    @include form-star-rating($brand-orange, $brand-orange);
   }
 
   &:hover input + label {
-    @include form-star-rating($white);
+    @include form-star-rating($white, $white);
   }
 
   > input:hover ~ label {
-    @include form-star-rating($gray-700);
+    @include form-star-rating($gray-500, $black);
   }
 
   > input:hover + label {
-    @include form-star-rating($white);
+    @include form-star-rating($white, $white);
   }
 }
 
-.star-rating-sm > label {
-  font-size: $form-star-size-sm;
+.star-rating-sm {
+
+  > label {
+    font-size: $form-star-size-sm;
+    @include form-star-rating-sm($accessible-orange, $accessible-orange);
+  }
+
+  > input:checked ~ label {
+    @include form-star-rating-sm($gray-700, $white);
+  }
+
+  > input:checked + label {
+    @include form-star-rating-sm($accessible-orange, $accessible-orange);
+  }
+
+  &:hover input + label {
+    @include form-star-rating-sm($black, $black);
+  }
+
+  > input:hover ~ label {
+    @include form-star-rating-sm($gray-700, $white);
+  }
+
+  > input:hover + label {
+    @include form-star-rating-sm($black, $black);
+  }
```

Comment:
Do we need all of those rules ?
Maybe yes for the maintainability.
Otherwise we might be able to remove the rules with filled stars.

Reply:
I removed as many common rules as possible: rules about diabled state or the common set of rules for label and span at the beggining.

It may be better to keep it like it is since we now have multiple set of colors depending of size variant, light/dark mode and state. It may be the only way to keep enough flexibility inside this component.